### PR TITLE
Remove isLegacyItCall of conformance tests

### PR DESCRIPTION
**Pull request by oomichi**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Today there are not any conformance tests which are added as
"[Confromance]" tag directly in the test names with the legacy
way. So this removes the corresponding code for cleanup.

**Does this PR introduce a user-facing change?**:

```NONE```

### CODE REVIEW

File: test/conformance/walk.go [hunk 1]
```diff
@@ -291,10 +259,6 @@ func (v *visitor) Visit(node ast.Node) (w ast.Visitor) {
 			totalConfTests++
 			v.emit(t.Args[0])
 			return nil
-		} else if v.isLegacyItCall(t) {
-			totalLegacyTests++
-			v.failf(t, "Using It() with manual [Conformance] tag is no longer allowed.  Use framework.ConformanceIt() instead.")
-			return nil
```
Comment (pohly) [inline test/conformance/walk.go]: The output still uses `totalLegacyTests`:
https://github.com/kubernetes/kubernetes/blob/6e3d21b7b8c6d1f6d6a615cc1e2c6a87f47e8391/test/conformance/walk.go#L329-L332

Should the variable and that output perhaps be removed completely?

**Comment by spiffxp** [comment]
I feel like there's value in keeping this check around to continue enforcing the requirement of `ConformanceIt` rather than placing this burden on reviewers for all future conformance tests.  It could maybe be rewritten to be less targeted toward cleanup, but it seems like a useful verify.

**Comment by oomichi** [comment]
@spiffxp Yeah, I can see the point. I checked the git history and the last test which used "[Conformance]" has been replaced since 26 Oct 2017 https://github.com/kubernetes/kubernetes/commit/a5967cbaf10e949aaa6e72ff6272fe018bcf2526
Maybe we need to keep it more time?

File: test/conformance/walk.go (hunk 1 above)
Comment (oomichi) [reply test/conformance/walk.go]: Oh, I missed that.
Yeah, that is a nice point.
I will do that.

**Comment by spiffxp** [comment]
I'd rather leave this check in.  I think we can drop the accounting part, but let's use this as an example of linting our conformance tests, and continue to fail if we find a test that doesn't follow convention.

**Comment by oomichi** [comment]
@spiffxp Thanks for your comment, that makes sense.
Yeah, it is nice to keep this check part for our future also.

**Pull request closed without merging**